TestPreShutdownHooks flakes in pull-kubernetes-unit

/kind flake
/sig api-machinery

TestPreShutdownHooks flakes a lot after this pr was merged.

- https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/batch/pull-kubernetes-unit/1508966938808684544
- https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/108492/pull-kubernetes-unit/1508972217449320448
- https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/batch/pull-kubernetes-unit/1508975494572281856
- https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/batch/pull-kubernetes-unit/1508981523284496384

_Originally posted by @pacoxu in https://github.com/kubernetes/kubernetes/pull/108033#issuecomment-1082531826_

```
{Failed  === RUN   TestPreShutdownHooks
    genericapiserver_graceful_termination_test.go:560: Sending request - timeout: 100ms, url: https://127.0.0.1:42179/echo?message=attempt-1
I0330 02:20:00.661251   78451 tlsconfig.go:240] "Starting DynamicServingCertificateController"
I0330 02:20:00.661763   78451 secure_serving.go:210] Serving securely on [::]:42179
    genericapiserver_graceful_termination_test.go:671: [server] seen new connection: &net.TCPConn{conn:net.conn{fd:(*net.netFD)(0xc000678100)}}
    genericapiserver_graceful_termination_test.go:653: [server] received a request, proto: HTTP/2.0, url: /echo?message=attempt-1
    genericapiserver_graceful_termination_test.go:543: The API server has started
I0330 02:20:00.727127   78451 object_count_tracker.go:84] "StorageObjectCountTracker pruner is exiting"
    genericapiserver_graceful_termination_test.go:560: Sending request - timeout: 100ms, url: https://127.0.0.1:42179/echo?message=attempt-0
    genericapiserver_graceful_termination_test.go:671: [server] seen new connection: &net.TCPConn{conn:net.conn{fd:(*net.netFD)(0xc0002a2000)}}
    genericapiserver_graceful_termination_test.go:560: Sending request - timeout: 100ms, url: https://127.0.0.1:42179/echo?message=attempt-1
    genericapiserver_graceful_termination_test.go:671: [server] seen new connection: &net.TCPConn{conn:net.conn{fd:(*net.netFD)(0xc0001b0900)}}
    genericapiserver_graceful_termination_test.go:560: Sending request - timeout: 100ms, url: https://127.0.0.1:42179/echo?message=attempt-2
    genericapiserver_graceful_termination_test.go:671: [server] seen new connection: &net.TCPConn{conn:net.conn{fd:(*net.netFD)(0xc0003c4700)}}
    genericapiserver_graceful_termination_test.go:560: Sending request - timeout: 100ms, url: https://127.0.0.1:42179/echo?message=attempt-3
    genericapiserver_graceful_termination_test.go:671: [server] seen new connection: &net.TCPConn{conn:net.conn{fd:(*net.netFD)(0xc000678480)}}
    genericapiserver_graceful_termination_test.go:560: Sending request - timeout: 100ms, url: https://127.0.0.1:42179/echo?message=attempt-4
    genericapiserver_graceful_termination_test.go:671: [server] seen new connection: &net.TCPConn{conn:net.conn{fd:(*net.netFD)(0xc0001b1080)}}
    genericapiserver_graceful_termination_test.go:478: PreSHutdown hook can not access the API server - Get "https://127.0.0.1:42179/echo?message=attempt-4": context deadline exceeded
--- FAIL: TestPreShutdownHooks (4.63s)
}
```

https://github.com/kubernetes/kubernetes/blob/0c5574c2a3211c3673ee01703c96a24cf58b4b4d/staging/src/k8s.io/apiserver/pkg/server/genericapiserver_graceful_termination_test.go#L446

--- a/staging/src/k8s.io/apiserver/pkg/server/genericapiserver_graceful_termination_test.go
+++ b/staging/src/k8s.io/apiserver/pkg/server/genericapiserver_graceful_termination_test.go
@@ -443,7 +443,7 @@ func TestPreShutdownHooks(t *testing.T) {
 		var r result
 		client := newClient(true)
 		for i := 0; i < 5; i++ {
-			r = doer.Do(client, func(httptrace.GotConnInfo) {}, fmt.Sprintf("/echo?message=attempt-%d", i), 100*time.Millisecond)
+			r = doer.Do(client, func(httptrace.GotConnInfo) {}, fmt.Sprintf("/echo?message=attempt-%d", i), 1*time.Second)
 			if r.err != nil {
 				break
 			}

